dll/win32/browseui/internettoolbar.cpp

```diff
@@ -1101,6 +1101,19 @@ HRESULT CInternetToolbar::ToggleBandVisibility(int BandID)
     else
         bandInfo.fStyle |= RBBS_HIDDEN;
 
+    if (BandID == ITBBID_TOOLSBAND)
+    {
+        int BtnCount = ::SendMessage(fToolbarWindow, TB_BUTTONCOUNT, 0, 0);
+
+        for (int idx = 0; idx < BtnCount; idx++)
+        {
+            TBBUTTONINFO BtnInfo = {sizeof(TBBUTTONINFO), TBIF_BYINDEX | TBIF_STATE};
+            ::SendMessage(fToolbarWindow, TB_GETBUTTONINFO, idx, (LPARAM)&BtnInfo);
+            BtnInfo.fsState = bandInfo.fStyle & RBBS_HIDDEN ? BtnInfo.fsState | TBSTATE_HIDDEN : BtnInfo.fsState & ~TBSTATE_HIDDEN;
```

Comment:
```suggestion
            BtnInfo.fsState = (bandInfo.fStyle & RBBS_HIDDEN) ? (BtnInfo.fsState | TBSTATE_HIDDEN) : (BtnInfo.fsState & ~TBSTATE_HIDDEN);
```
or in case this actually makes the code more readable:
```suggestion
            if (bandInfo.fStyle & RBBS_HIDDEN)
                BtnInfo.fsState |= TBSTATE_HIDDEN
            else
                BtnInfo.fsState &= ~TBSTATE_HIDDEN;
```